projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,432 @@
+7. `mint_authority_address` - (Optional) Address to send the mint authority output
+8. `allow_external_mint_authority_address` - (Defaults to false) Flag indicating if the mint authority address is allowed to be from another wallet. 
+9. `create_melt` - (Defaults to true) If the wallet should create a melt authority for the newly created token
+10. `melt_authority_address` - (Optional) Address to send the melt authority output
+11. `allow_external_melt_authority_address` - (Defaults to false) Flag indicating if the melt authority address is allowed to be from another wallet.
+
+**Example Request:**
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_signWithAddress",
+  "params": {
+	"name": "Bathor",
+	"symbol": "BTR",
+	"amount": 500,
+	"address": "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c",
+	"create_mint": false,
+	"mint_authority_address": "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c",
+	"create_melt": false,
+	"melt_authority_address": "H8RmX1AMQKAhWkBvf8DaKoAU7ph3Yiqg3c"
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "txId": "00004788e55b9b3fb90aaad2065e99ccf36ccd2fac5ed40ae62eeefd2486e96c",
+  "name": "Bathor",
+  "symbol": "BTR",
+  "amount": 500
+}
+```
+
+
+### 2.1.3 `htr_getUtxos`
+
+The `htr_getUtxos` method will search for `utxos`  given a set of filters
+
+**Parameters**
+
+1. `maxUtxos` - (Defaults to 255) Maximum number of `utxos` to return.
+2. `token` - (Defaults to HTR) Token to filter the `utxos`
+3. `filterAddress` - Address to filter the `utxos`. The address must belong to the initialized wallet
+4. `amountSmallerThan` - (Optional) Maximum limit of `utxo` amount to filter the list. We will only return `utxos` that have the `amount` lower than this value. Integer representation of decimals, i.e. 100 = 1.00.
+5. `amountBiggerThan` - (Optional) Minimum limit of `utxo` amount to filter the list. We will only return `utxos` that have the `amount` higher than this value. Integer representation of decimals, i.e. 100 = 1.00.
+6. `maximumAmount` - (Optional) Limit the maximum total amount to return summing all utxos. Integer representation of decimals, i.e. 100 = 1.00.
+7. `onlyAvailableUtxos` - (Defaults to true) Get only available `utxos`, ignoring locked ones.
+
+**Example Request:**
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_getUtxos",
+  "params": {
+	"maxUtxos": 50,
+	"token": "BTR",
+	"amount": 1
+  }
+}
+```
+
+**Example Response:**
+
+```json
+{
+  "total_amount_available": 5,
+  "total_utxos_available": 10,
+  "total_amount_locked": 0,
+  "total_utxos_locked": 0,
+  "utxos": [
+    {
+      "address": "HNnK9wgUVL6Cjzs1K3jpoGgqQTXCqpAnW8",
+      "amount": 1,
+      "tx_id": "00fff7a3c6eb95ec3343bffcfca9a3a0d3e243462ae7de1f200cdd76716140fb",
+      "locked": false,
+      "index": 0
+    }
+  ]
+}
+```
+
+
+### 2.1.4 `htr_signWithAddress`
+
+The `htr_signWithAddress` method requests a signed message using an address' private key that can be verified.
```

Comment:
I think we should add that the message will receive a fixed prefix before signing.

Reply:
Done in 330bc6059de95fce7a5364b6a8cca197919daa51

Reply:
Great. Thanks! I think we can add an example. For instance:

```
message = "This is my message"
signed_message = "Hathor Signed Message: This is my message"
```

Reply:
Great! Thanks!